server/src/service-layer/controllers/UserSignup.ts

```diff
@@ -0,0 +1,52 @@
+import AuthService from "business-layer/services/AuthService"
+import UserDataService from "data-layer/services/UserDataService"
+import { UserRecord } from "firebase-admin/auth"
+import { UserSignupBody } from "service-layer/request-models/UserSignupRequests"
+import { Body, Controller, Post, Route, SuccessResponse } from "tsoa"
+
+@Route("signup")
+export class UserSignup extends Controller {
+  @Post()
+  @SuccessResponse(200)
+  // return a JWT token at the end
+  public async signup(@Body() requestBody: UserSignupBody): Promise<string> {
+    // create services
+    const userService = new UserDataService()
+    const authService = new AuthService()
+
+    let user: UserRecord
+    console.log(requestBody) // debug
+    try {
+      // create user in auth service
+      // const user = await authService.createUser(requestBody.email)
+      user = await authService.createUser(requestBody.email)
```

Comment:
Might want to destructure here
```suggestion
      { uid } = await authService.createUser(requestBody.email)
```